core/dbt/parser/base.py

```diff
@@ -314,6 +314,7 @@ def update_parsed_node_config(
         # compatibility with earlier node-only config.
         if config_dict.get("contract", False):
             parsed_node.contract = True
+            parsed_node.build_contract_checksum()
```

Comment:
This should be called once the `parsed_node` has `columns`, since the contract checksum is derived from the columns. Otherwise `parsed_node.columns` will always be `{}` and produce the same checksum. I think that's what's currently happening. Perhaps we could call `build_contract_checksum` [here](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/parser/schemas.py#L940) instead, after the config and columns are patched, and constraints are validated.